File: VB/GridDemo.Wpf/Modules/EntityFrameworkServerMode/EntityFrameworkInstantFeedbackModeViewModel.vb
Imports DevExpress.Xpf.DemoBase
Imports System.Collections.Generic
Imports System.Data.SQLite

Namespace GridDemo
    Public Class EntityFrameworkInstantFeedbackModeViewModel
        Inherits InstantFeedbackModeViewModelBase(Of OutlookDataRecord)

        #Region "DatabaseInfo"
        Public Shared ReadOnly DatabaseInfo As New DatabaseInfo(OutlookDataContext.FileName, "OutlookDataRecords", GetType(OutlookDataRecord), AddressOf CreateValues, Function(sql, connection) New SQLiteCommand(sql, CType(connection, SQLiteConnection)))
        Private Shared Function CreateValues() As Dictionary(Of String, Object)
            Dim subject = OutlookDataGenerator.GetSubject()
            Dim hasAttachment = OutlookDataGenerator.GetHasAttachment()
            Dim size = OutlookDataGenerator.GetSize(hasAttachment)
            Dim [from] = OutlookDataGenerator.GetFrom()
            Dim sent = OutlookDataGenerator.GetSentDate()
            Dim dict = New Dictionary(Of String, Object)()
            dict.Add("Subject", subject)
            dict.Add("HasAttachment", hasAttachment)
            dict.Add("Size", size)
            dict.Add("From", [from])
            dict.Add("Sent", sent)
            Return dict
        End Function
        #End Region

        Protected Sub New()
            MyBase.New(DatabaseInfo, Function() (New OutlookDataContext()).OutlookDataRecords)
        End Sub
    End Class
End Namespace

File: CS/GridDemo.Wpf/Modules/EntityFrameworkServerMode/EntityFrameworkInstantFeedbackModeViewModel.cs
using DevExpress.Xpf.DemoBase;
using System.Collections.Generic;
using System.Data.SQLite;

namespace GridDemo {
    public class EntityFrameworkInstantFeedbackModeViewModel : InstantFeedbackModeViewModelBase<OutlookDataRecord> {
        #region DatabaseInfo
        public static readonly DatabaseInfo DatabaseInfo = new DatabaseInfo(
            OutlookDataContext.FileName, "OutlookDataRecords", typeof(OutlookDataRecord), CreateValues, (sql, connection) => new SQLiteCommand(sql, (SQLiteConnection)connection));
        static Dictionary<string, object> CreateValues() {
            var subject = OutlookDataGenerator.GetSubject();
            var hasAttachment = OutlookDataGenerator.GetHasAttachment();
            var size = OutlookDataGenerator.GetSize(hasAttachment);
            var from = OutlookDataGenerator.GetFrom();
            var sent = OutlookDataGenerator.GetSentDate();
            var dict = new Dictionary<string, object>();
            dict.Add("Subject", subject);
            dict.Add("HasAttachment", hasAttachment);
            dict.Add("Size", size);
            dict.Add("From", from);
            dict.Add("Sent", sent);
            return dict;
        }
        #endregion

        protected EntityFrameworkInstantFeedbackModeViewModel()
            : base(DatabaseInfo, () => new OutlookDataContext().OutlookDataRecords) {
        }
    }
}
